lib/player.js

```diff
@@ -3891,9 +3939,30 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
 
     this.manifest_.textStreams.push(stream);
     this.onTracksChanged_();
-    return shaka.util.StreamUtils.textStreamToTrack(stream);
+    const track = shaka.util.StreamUtils.textStreamToTrack(stream);
+    return Promise.resolve(track);
```

Comment:
In an async function, you don't need to use Promise.resolve, either.  The Promise is implicit in an async function.  Just return the value of track, and it will be used to resolve the implied Promise.